### DISCUSSION TITLE
The compiler should warn when ";" is accidentally used instead of ":" at the end of a line

### DISCUSSION THREAD

**Discussion by smarter** [scala/scala3#18777] [Feature Requests]
Spot the bug:
```scala
object A;
  def foo = 1

A.foo // error: value foo is not a member of A
```
On a French keyboard, `;` and `:` are on adjacent keys, so this is an easy mistake to make and hard to spot given how similar both of these characters look. When omitting the `:` we get a warning, but no such warning appears when the line ends with a `;`:
```scala
object A
  def foo = 1
```
```
-- Warning: --------------------------------------------------------------------
2 |  def foo: Int = 1
  |  ^
  |  Line is indented too far to the right, or a `{` or `:` is missing
```

**Comment by som-snytt** [comment]
It could lint any semi that is not required for parsing. The optional semi before a guard in a for loop may be controversial enough to occasion fisticuffs in the break room.
